fs/test/tests.go

```diff
@@ -28,10 +28,137 @@ package test
 // this file implements utility routines related to testing.
 
 import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
 	"path/filepath"
 	"testing"
 )
 
+// readAndRestoreError records an error for restoring the *os.file handle.
+type readAndRestoreError struct{}
+
+// Error returns an error when restoring an already closed handle.
+func (r *readAndRestoreError) Error() string {
+	return "ReadAndRestore from closed MockStdInErr"
+}
+
+// MockStdInErr represents a mock implementation of STDIN and STDERR.
+type MockStdInErr struct {
+	origStdin    *os.File
+	stdinWriter  *os.File
+	origStderr   *os.File
+	stderrReader *os.File
+	outCh        chan []byte
+}
+
+// On creation of a new MockStdInErr, it writes the given text to STDIN, and
+// starts capturing STDERR. Be sure to call RestoreStdIn() after you've done
+// reading from STDIN.
+func NewMockStdInErr(stdinText string) (*MockStdInErr, error) {
```

Comment:
For times when we only want to capture stderr, there should be a 'NewMockStdErr', so callers don't have to remember to call RestoreStdIn() when they weren't even doing anything with STDIN. That method can be called here and the stdin stuff added in to the returned struct.